### ISSUE TITLE
DRAGMAP_Alignment_error

### ISSUE THREAD

**Issue by poddarharsh15** [nf-core/sarek#1555]
### Description of the bug

[my_param.json](https://github.com/user-attachments/files/15766958/my_param.json)

> >  error [nextflow.exception.ProcessFailedException]: Process `NFCORE_SAREK:SAREK:FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP_SENTIEON:DRAGMAP_ALIGN (sample1-lane_1)` terminated with an error exit status (1)
> Jun-10 12:48:18.909 [Task submitter] DEBUG n.executor.local.LocalTaskHandler - Launch cmd line: /bin/bash -ue .command.run
> Jun-10 12:48:18.909 [Task submitter] INFO  nextflow.Session - [5d/c27009] Submitted process > NFCORE_SAREK:SAREK:FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP_SENTIEON:DRAGMAP_ALIGN (sample1-lane_1)
> Jun-10 12:48:18.917 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'NFCORE_SAREK:SAREK:FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP_SENTIEON:DRAGMAP_ALIGN (sample1-lane_1)'
> 
> Caused by:
>   Process `NFCORE_SAREK:SAREK:FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP_SENTIEON:DRAGMAP_ALIGN (sample1-lane_1)` terminated with an error exit status (1)
> 
> Command executed:
> 
>   dragen-os \
>       -r dragmap \
>       --RGSM patient1_sample1 --RGID "@RG\tID:H8VDAADXX.sample1.lane_1\tPU:lane_1\tSM:patient1_sample1\tLB:sample1\tDS:s3://ngi-igenomes/igenomes//Homo_sapiens/GATK/GRCh38/Sequence/WholeGenomeFasta/Homo_sapiens_assembly38.fasta\tPL:ILLUMINA" \
>       --num-threads 12 \
>       -1 0002.sample1-lane_1_1.fastp.fastq.gz -2 0002.sample1-lane_1_2.fastp.fastq.gz \
>       2> >(tee sample1-lane_1.0002.dragmap.log >&2) \
>       | samtools sort -n --threads 12  -o sample1-lane_1.0002.bam -
>   
>   cat <<-END_VERSIONS > versions.yml
>   "NFCORE_SAREK:SAREK:FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP_SENTIEON:DRAGMAP_ALIGN":
>       dragmap: $(echo $(dragen-os --version 2>&1))
>       samtools: $(echo $(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*$//')
>       pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )
>   END_VERSIONS
> 
> Command exit status:
>   1
> 
> Command output:
>   (empty)
> 
> Command error:
>   When maskLen < 15, the function ssw_align doesn't return 2nd best alignment information.
>   When maskLen < 15, the function ssw_align doesn't return 2nd best alignment information.
>   When maskLen < 15, the function ssw_align doesn't return 2nd best alignment information.
>   When maskLen < 15, the function ssw_align doesn't return 2nd best alignment information.
>   When maskLen < 15, the function ssw_align doesn't return 2nd best alignment information.

### Command used and terminal output

```console
nextflow run nf-core/sarek -r 3.4.2 \
-profile singularity \
-params-file my_param.json \
--outdir results_sarek_2 \
-resume
```

### Relevant files

[nextflow.log](https://github.com/user-attachments/files/15767160/nextflow.log)

**Comment by FriederikeHanssen** [comment]
This looks like it might be direct error with dragmap. Did you check with their developers if there is anything particular about your data and the way it is used?

**Comment by poddarharsh15** [comment]
> This looks like it might be direct error with dragmap. Did you check with their developers if there is anything particular about your data and the way it is used?

Yes, the problem is with dragmap, but they haven't fixed it yet. However, we are working on it with the help of @edmundmiller  [here](https://nfcore.slack.com/archives/C05V9FRJYMV/p1720105679738269?thread_ts=1719221876.617059&cid=C05V9FRJYMV) and you can also follow the progress on GitHub [here](https://github.com/gambalab/DRAGMAP/commit/e7fdb6c68b52f7f192bdae2ec1e5720fc7c7a81d). hopefully it will be solved soon.

**Comment by edmundmiller** [comment]
1.4.2 should be on bioconda!

**Comment by edmundmiller** [comment]
@poddarharsh15 https://github.com/nf-core/modules/pull/4895 if you want to add your test case here!

**Comment by poddarharsh15** [comment]
> @poddarharsh15 [nf-core/modules#4895](https://github.com/nf-core/modules/pull/4895) bumped the images here if you want to add your test case!

Hi @edmundmiller sorry for this question but I didn't understand exactly what do mean by bumping the image, the image from  this link [https://quay.io/repository/biocontainers/narfmap?tab=tags] or the one I used earlier for running test??

**Comment by poddarharsh15** [comment]
> The one from the link. It has the static compilation fix in `1.4.1` and then `1.4.2` which fixed the bioconda compile error.

I understand what you're suggesting, but can't I use the same module that you're updating? In the end, I will need dragmap working with the nf-core/sarek pipeline for patient data inspection. Please let me know if it's possible?

**Comment by edmundmiller** [comment]
In theory, you can, eventually the repos will diverge. Should work for your purposes, though!

**Comment by poddarharsh15** [comment]
> In theory, you can, eventually the repos will diverge. Should work for your purposes, though!

Are you going to diverge these repos in `dragmap` modules? Or introducing a new module under your repos?

**Comment by edmundmiller** [comment]
Late response, planning on introducing a new module under the narfmap repo to denote that it's not the same project to avoid confusing anyone.

**Comment by YasirKusay** [comment]
I had a similar issue. I switched over to conda and changed the dragmap environment.yaml file to use the latest versions of dragmap and samtools, unfortunately there is a different issue now (Segmentation fault)
